homeassistant/components/modbus/sensor.py

```diff
@@ -209,16 +222,16 @@ def __init__(
         self._register_type = entry[CONF_INPUT_TYPE]
         self._unit_of_measurement = entry.get(CONF_UNIT_OF_MEASUREMENT)
         self._count = int(entry[CONF_COUNT])
-        self._swap = entry[CONF_SWAP]
         self._scale = entry[CONF_SCALE]
         self._offset = entry[CONF_OFFSET]
         self._precision = entry[CONF_PRECISION]
-        self._structure = structure
         self._data_type = entry[CONF_DATA_TYPE]
         self._device_class = entry.get(CONF_DEVICE_CLASS)
         self._value = None
-        self._available = True
         self._scan_interval = timedelta(seconds=entry.get(CONF_SCAN_INTERVAL))
+        self._structure, self._swap = ModbusRegisterSensor._create_data_structure(entry)
```

Comment:
Downside with this is that we now produce side effects in the form of logging messages in the init method. We try to avoid side effects in init methods.

Reply:
we do already have a message if the structure is false.